python/paddle/fluid/input.py

```diff
@@ -115,6 +116,18 @@ def embedding(input,
     """
 
     helper = LayerHelper('embedding', **locals())
+    if not isinstance(input, Variable):
+        raise TypeError(
+            "The type of 'input' in fluid.embedding must be Variable, but received %s"
+            % (type(input)))
+    if convert_dtype(input.dtype) not in ['int64']:
+        raise TypeError(
+            "The data type of 'input' in fluid.embedding must be int64, but received %s."
+            % (convert_dtype(input.dtype)))
+    if convert_dtype(dtype) not in ['float32', 'float64']:
```

Comment:
dtype参数的文档列举的类型和这里代码不太一致，确认一下是否无误
```
        dtype(np.dtype|core.VarDesc.VarType|str): The dtype refers to the data type of output
            tensor. It can be float32, float_16, int etc.
```

Reply:
已在.h和.cu文件里修改

Reply:
搭了简单的网络，np.type目前是已经不支持了，但只支持core.VarDesc.VarType。文档会单独提PR修改这个。后期会关联这个PR

Reply:
英文文档PR：https://github.com/PaddlePaddle/Paddle/pull/20187